public/app/models/query/Group.js

```diff
@@ -0,0 +1,90 @@
+// Based on http://kindohm.github.io/knockout-query-builder/js/group.js
+(function () {
+    'use strict';
+
+    define(
+        [
+            'lodash',
+            'knockout',
+            'models/query/Condition'
+        ],
+        function (_, ko, Condition) {
+            var Group;
+
+            Group = function (queryBuilder, parentGroup) {
+                this.children = ko.observableArray();
+                this.selectedOperator = ko.observable();
+
+                this.operators = ko.pureComputed(function () {
+                    return [
+                        {
+                            label: 'and',
+                            value: 'AND'
+                        },
+                        {
+                            label: 'or',
+                            value: 'OR'
+                        }
+                    ];
+                });
+
+                this.depth = ko.pureComputed(function () {
+                    return parentGroup ? 1 + parentGroup.depth() : 0;
+                });
+
+                this.allowChildren = ko.pureComputed(function () {
+                    return this.depth() + 1 < queryBuilder.maxDepth();
+                }.bind(this));
+
+                this.query = ko.pureComputed(function () {
+                    return {
+                        operator: this.selectedOperator(),
+                        children: _(this.children()).invokeMap('query').filter().value()
+                    };
+                }.bind(this));
+
+                this.parse = function (query) {
```

Comment:
Again, inverse of `query`.  Takes the query structure and populates this model object based on the contents.